Content.Shared/Inventory/InventorySystem.Slots.cs

```diff
@@ -109,9 +111,18 @@ public bool MoveNext([NotNullWhen(true)] out ContainerSlot? container)
             container = null;
             if (_nextIdx >= _slots.Length) return false;
 
-            while (_nextIdx < _slots.Length && !_inventorySystem.TryGetSlotContainer(_uid, _slots[_nextIdx++].Name, out container, out _)) { }
+            while (_nextIdx < _slots.Length)
+            {
+                var slot = _slots[_nextIdx++];
```

Comment:
```suggestion
            for (; _nextIdx < _slots.Length; _nextIdx++)
            {
                var slot = _slots[_nextIdx];
```

Reply:
enumerator returns in the middle of the loop, so the increment from the for loop never happens.
reverting to a while loop with explicit increment.